rust/ql/test/library-tests/sensitivedata/test.rs

```diff
@@ -0,0 +1,150 @@
+
+fn get_string() -> String { "string".to_string() }
+
+fn sink<T>(_: T) { }
+
+// --- tests ---
+
+struct MyStruct {
+	harmless: String,
+	password: String,
+	password_file_path: String,
+	password_enabled: String,
+}
+
+impl MyStruct {
+	fn get_certificate(&self) -> String { return get_string() }
+	fn get_certificate_url(&self) -> String { return get_string() }
+	fn get_certificate_file(&self) -> String { return get_string() }
+}
+
+fn get_password() -> String { get_string() }
+
+fn test_passwords(
+	password: &str, passwd: &str, my_password: &str, password_str: &str, pass_phrase: &str,
+	auth_key: &str, authenticationkey: &str, authKey: &str,
+	harmless: &str, encrypted_password: &str, password_hash: &str,
+	ms: &MyStruct
+) {
+	// passwords
+	sink(password); // $ sensitive=password
+	sink(passwd); // $ sensitive=password
+	sink(my_password); // $ sensitive=password
+	sink(password_str); // $ sensitive=password
+	sink(pass_phrase); // $ MISSING: sensitive=password
+	sink(auth_key); // $ MISSING: sensitive=password
```

Comment:
should we add those?

we should also consider that snake-case is the convention in rust for variables (by default you even have that flagged and many IDEs propose to fix the case automatically), so I think we should add snakecase counterparts to the camelcase heuristics (`auth_key` and `authentication_key`).

Also, if we add `pass_phrase`/`passPhrase`, we should probably also add `passphrase`.

Reply:
> should we add those?

Yes we should, but I'd like to do it in the shared library and that requires agreement from other language teams (and wider testing).  So it'll be a follow-up PR, and it is planned.

> if we add pass_phrase/passPhrase, we should probably also add passphrase.

The regexs are all case insensitive, so `passPhrase` and `passphrase` should both be matched.  I can add some test cases around this if you like?

Reply:
I'm going to hit merge on this PR now.  As I said above, I plan to do more work on the sensitive data library in the near future so I'm more than happy to take further suggestions / feedback.

Reply:
I should add, you're 100% right that the frequency of `snake_case` is a quirk of Rust that I should be paying more attention to.